src/Haskell/Ide/Engine/Server.hs

```diff
@@ -152,10 +152,23 @@ run scheduler _origDir plugins captureFp = flip E.catches handlers $ do
           (Debounce.forMonoid $ react . dispatchDiagnostics)
           (Debounce.def { Debounce.delay = debounceDuration, Debounce.alwaysResetTimer = True })
 
+
+        let lspRootDir = Core.rootPath lf
+        currentDir <- liftIO getCurrentDirectory
+
+        -- Check for mismatching GHC versions
+        -- Ignore hie.yaml parse errors. They get reported in ModuleCache.hs
+        let parseErrorHandler (_ :: Yaml.ParseException) = do
+              logm "Caught a yaml parse exception"
+              return Nothing
+            dummyCradleFile = fromMaybe currentDir lspRootDir </> "File.hs"
+        logm $ "Dummy Cradle File: " ++ dummyCradleFile
+        mcradle <- liftIO $ E.catch (Just <$> findLocalCradle dummyCradleFile) parseErrorHandler
```

Comment:
Should the check in `NotInitialized` be moved here or can we pass the found cradle somehow to `onStartup`? Or the other way around?

Reply:
We should keep the check in NotInitialized as it's really the only appropriate time to send the `showMessage` request to the user. We probably don't need to log anything here then, just swallow the exception. 
As a side note, should `findCradle` in hie-bios not do this exception handling for us, returning a `Maybe` instead?

Reply:
The best time to send the messages is in `onStartup`. `NotInitialized` is just the final handshake from the client that it is done.  `onStartup` is our internal processor of the `initialize` request from the client.

Reply:
I would prefer it in `onStartup` also for DRY reasons. And if it is like @alanz says, I think it is just as valid and reasonable.